### DISCUSSION TITLE
[Feature] Add docker tag for minor versions (e.g. v1.78) that points at most recent patch (e.g. v1.78.1)

### DISCUSSION THREAD

**Discussion by deafmute1** [immich-app/immich#4149] [Feature Request]
### The feature

So this means you have these two containers with the following tags:
Container 1: v1.78.1, **v1.78**, release
Container 2: v1.78.0
This is a common feature that docker containers have. This means I can get hotfixes/patches without needing to check release notes or do any manual migrations. Personally my current upgrade model for docker containers is: some containers that i want to manually upgrade major versions or otherwise supervise I have specific major/minor version tag; other containers I just set to release/latest. This means I can watchtower without manual supervision with low risk, while picking up hotfixes. Then from time to time I come and supervise the containers that need it on their major/minor versions. As it is right now, I often don't get hotfixes for immich because I cannot point to the latest patch of a minor version. This somewhat negates the point of their existence.

### Platform

- [x] Server
- [ ] Web
- [ ] Mobile

**Comment by draevin** [comment]
I see #22556 was closed as a dupe of this one, so I assume (since it isn't mentioned here) that a resolution for this would include major versions to latest minor+patch as well; is that correct?

**Comment by rajalah71** [comment]
Major (and major.minor) tags would be greatly appriciated especially now that we have a stable release.

**Comment by DonkeeeyKong** [comment]
I'll add the text of my original feature request https://github.com/immich-app/immich/discussions/22556, that was closed in favor of this one here:

Congratulations to the stable release! It's great news, that you plan to follow semantic versioning. May I suggest to start using a similar approach to docker image tags as many other projects that use semantic versioning e.g., Postgres or [Valkey](https://hub.docker.com/r/valkey/valkey)? In a way that release `2.0.0` is tagged with `2`, `2.0` and `2.0.0,` while `2.0.1` has the tags `2`, `2.0` and `2.0.1`, `2.5.3` has the tags `2,` `2.5` and `2.5.3`, etc.?

With semantic versioning not having planned breaking changes within a major version, this way users would be able to set `IMMICH_VERSION` to `2` and auto-pull the images. This would be a great improvement!

**Comment by ElioDiNino** [comment]
This is now partially resolved with v2.1.0! There are tags for v2 (new) and v2.1.0 (existing pattern). The associated change was [here](https://github.com/immich-app/devtools/commit/2f0961babab188fdde832563ad9ebc0baee6b89b). Maybe @bo0tzz would consider adding the minor version tag as well?

**Reply by bo0tzz** [reply]
Our minor trains are usually pretty short, so I don't really see a point to that.

**Reply by ldobbelsteen** [reply]
But what if there are any deployment changes between minor versions? Like an update to the Valkey or Postgres images? I think the point of these granular tags it to be able to auto-update safely without worrying about deployment-based breakage.

Unless you will aim for not doing deployment changes until v3

**Reply by bo0tzz** [reply]
> But what if there are any deployment changes between minor versions? 

There won't be any breaking changes, that's the point of being stable and using semver.

> Like an update to the Valkey or Postgres images?

These are routine.

**Reply by Magicrafter13** [reply]
thank you all for the `v2` (and future) tags, it is extremely appreciated
